src/uu/numfmt/src/errors.rs

```diff
@@ -0,0 +1,34 @@
+use std::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+use uucore::error::UError;
+
+#[derive(Debug)]
+pub enum NumfmtError {
+    IoError(String),
+    IllegalArgument(String),
+    FormattingError(String),
+}
+
+impl UError for NumfmtError {
+    fn code(&self) -> i32 {
+        match self {
+            NumfmtError::IoError(_) => 1,
```

Comment:
it is going to be covered by a test ?

Reply:
I can try, is there a good example of how to consistently inject an IO error into stdin somewhere?

Reply:
not sure, you should look elsewhere, sorry